### DISCUSSION TITLE
sproto解析时遇见一些奇怪的问题，报错内容难懂，希望大佬们给点建议。

### DISCUSSION THREAD

**Discussion by LostCompanies** [cloudwu/skynet#1932] [Q&A] [answered]
项目中传递sproto第一次解析无问题，第二次同样的消息传递进来，就会报错。一头雾水。
但是加了print后，报错消失。同时过段时间，程序崩溃报错 Segmentation fault。
请问有大佬给点解决方向吗？
初学者，目前只是在测试阶段。

./lualib/skynet.lua:858: ./lualib/skynet.lua:330: ./lualib/skynet/queue.lua:20: ./lualib/sproto.lua:193: bad argument #2 to 'encode' (nil expected, got table)

**Comment by LostCompanies** [comment]
或者有没有好用的调试工具介绍呢？目前是开了虚拟机跑

**Comment by CoolDesert** [comment]
大多数 issue 里的 skynet 奔溃问题，最终都是自己写的 C 库导致，可以打开 MEMORY_CHECK 宏看看是否有越界或 double free

**Comment by LostCompanies** [comment]
经过不断试错，发现
local message = netpack.tostring(msg, sz) 出bug
local message = skynet.tostring(msg, sz) 不会

**Reply by cloudwu** [reply, marked as answer]
netpack 并非公开 api ，仅供内部实现使用。

不要使用它。

https://github.com/cloudwu/skynet/wiki/GateServer  注意文档中强调：**除非你认为已经了解了细节和具备出错调试的能力，否则请不要直接使用 netpack 。**

**Reply by LostCompanies** [reply]
感谢大佬回复。这就去再读一遍文档(^v^)

**Comment by qhf566** [comment]
谁有大厂,近5年内开发完的,国内没有运营过的,成品开发完毕的手机游戏,联系我809996677
